operator/pkg/reconciliation/decommission_node.go

```diff
@@ -0,0 +1,340 @@
+		rackInfo := rc.desiredRackInformation[idx]
+		statefulSet := rc.statefulSets[idx]
+		desiredNodeCount := int32(rackInfo.NodeCount)
+		maxReplicas := *statefulSet.Spec.Replicas
+		lastPodSuffix := stsLastPodSuffix(maxReplicas)
+
+		if maxReplicas > desiredNodeCount {
+			dcPatch := client.MergeFrom(dc.DeepCopy())
+			updated := false
+
+			updated = rc.setCondition(
+				api.NewDatacenterCondition(
+					api.DatacenterScalingDown, corev1.ConditionTrue)) || updated
+
+			if updated {
+				err := rc.Client.Status().Patch(rc.Ctx, dc, dcPatch)
+				if err != nil {
+					logger.Error(err, "error patching datacenter status for scaling down rack started")
+					return result.Error(err)
+				}
+			}
+
+			rc.ReqLogger.Info(
+				"Need to update the rack's node count",
+				"Rack", rackInfo.RackName,
+				"maxReplicas", maxReplicas,
+				"desiredSize", desiredNodeCount,
+			)
+
+			rc.Recorder.Eventf(rc.Datacenter, corev1.EventTypeNormal, events.ScalingDownRack,
+				"Scaling down rack %s", rackInfo.RackName)
+
+			if err := setOperatorProgressStatus(rc, api.ProgressUpdating); err != nil {
+				return result.Error(err)
+			}
+
+			err := rc.DecommissionNodeOnRack(rackInfo.RackName, epData, lastPodSuffix)
+			if err != nil {
+				return result.Error(err)
+			}
+
+			return result.RequeueSoon(10)
+		}
+	}
+
+	return result.Continue()
+}
+
+func (rc *ReconciliationContext) DecommissionNodeOnRack(rackName string, epData httphelper.CassMetadataEndpoints, lastPodSuffix string) error {
+	for _, pod := range rc.dcPods {
+		podRack := pod.Labels[api.RackLabel]
+		if podRack == rackName && strings.HasSuffix(pod.Name, lastPodSuffix) {
+			mgmtApiUp := isMgmtApiRunning(pod)
+			if !mgmtApiUp {
+				return fmt.Errorf("Management API is not up on node that we are trying to decommission")
+			}
+
+			if err := rc.EnsurePodsCanAbsorbDecommData(pod, epData); err != nil {
+				return err
+			}
+
+			if err := rc.NodeMgmtClient.CallDecommissionNodeEndpoint(pod); err != nil {
+				// TODO this returns a 500 when it works
+				// We are waiting for a new version of mgmt api with a fix for this
+				// return false, err
+			}
+
+			rc.ReqLogger.Info("Marking node as decommissioning")
+			patch := client.MergeFrom(pod.DeepCopy())
+			pod.Labels[api.CassNodeState] = stateDecommissioning
+			if err := rc.Client.Patch(rc.Ctx, pod, patch); err != nil {
+				return err
+			}
+
+			rc.Recorder.Eventf(rc.Datacenter, corev1.EventTypeNormal, events.LabeledPodAsDecommissioning,
+				"Labeled node as decommissioning %s", pod.Name)
+
+			return nil
+		}
+	}
+
+	// this shouldn't happen
+	return fmt.Errorf("Could not find pod to decommission on rack %s", rackName)
+}
+
+// Wait for decommissioning nodes to finish before continuing to reconcile
+func (rc *ReconciliationContext) CheckDecommissioningNodes(epData httphelper.CassMetadataEndpoints) result.ReconcileResult {
+	if rc.Datacenter.GetConditionStatus(api.DatacenterScalingDown) == corev1.ConditionFalse {
```

Comment:
What if the condition is Unknown?

Reply:
Naturally I checked all of the cases in reconcile rack but not in my own file